subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DefaultFileSystemSnapshotterTest.groovy

```diff
@@ -243,6 +248,39 @@ class DefaultFileSystemSnapshotterTest extends Specification {
         })
     }
 
+    def "snapshots a singletonFileTree as RegularFileSnapshot"() {
+        given:
+        def file = tmpDir.createFile('testFile')
+
+        when:
+        def tree = new FileTreeAdapter(new DefaultSingletonFileTree(file), TestFiles.patternSetFactory)
+        def snapshots = snapshotter.snapshot(tree)
+
+        then:
+        snapshots.size() == 1
+        getSnapshotInfo(snapshots[0]) == [null, 1]
+    }
+
+    def "snapshots a generated singletonFileTree as RegularFileSnapshot"() {
```

Comment:
If we support deeper down relative paths, would the snapshot for `dir/file.txt` be then a `DirectorySnapshot`?

Reply:
I assumed it would also just be a RegularFileSnapshot

Reply:
Then the relative path of the root file would be wrong and we would not detect changes there. Let's get rid of relative path in `GeneratedSingletonFileTree`.

Reply:
removed relative path from `GeneratedSingletonFileTree`